### PR TITLE
OpenTNSim OpenCLSim connection

### PR DESCRIPTION
**Pull request by SiggyF**
- [x] Rename routeable -> routable (opentnsim)
- [x] Add assumptions for env (`.graph` (preferred) or `.FG`)
- [x] Add typed arguments for destination (shapely.Geometry, should be along the route for routables), engine_order (float) and duration (overrides v based duration computation).
- [x] Make types compatible between opentnsim and opentlsim
   - [x]  add optional node to locatable
   - [x] support both openclsim and opentnsim compatible logging
- [x] Split routable based moving vs non routable moving (always over Linestring), routable determines linestring, simple sailing is over linestring (source, destination)
- [x] Support routing over geometries (geometries as network nodes)
- [x] Explicitly define data model for nx.Graph to be able to route over it

(commits moved from https://github.com/TUDelft-CITG/OpenCLSim/pull/116)

### CODE REVIEW

File: src/openclsim/model/move_activity.py [hunk 1]
```diff
@@ -81,6 +81,8 @@ def main_process_function(self, activity_log, env):
             speed the mover should sail. for example, engine_order=0.5
             corresponds to sailing at 50% of max speed
         """
+
+        print("mover", self.mover.name, "destination", self.destination.name)
```
Comment (Pietervanhalem) [inline src/openclsim/model/move_activity.py:85]: remaining dev statement? If you want to keep it is should be done with the logger, but I preffer not

File: src/openclsim/model/move_activity.py [hunk 2]
```diff
@@ -118,3 +121,5 @@ def main_process_function(self, activity_log, env):
         self._release_resource(
             self.requested_resources, self.mover.resource, self.keep_resources
         )
+
+        print("mover", self.mover.name, "is now at", self.mover.geometry.wkt)
```
Comment (Pietervanhalem) [inline src/openclsim/model/move_activity.py:125]: same

**Review by Pietervanhalem — DISMISSED** [review]

**Comment by luke-moth** [comment]
@SiggyF 
looks good :)

- zojuist review besproken. Zie nog twee TODOS (code smells) en de falende tests hierboven. Daarnaast notebooks bekeken en los van comments in notebook 33 geen commentaar, zijn duidelijk/volgens structuur OpenCLSIm notebooks.

File: src/openclsim/model/move_activity.py (hunk 1 above)
Comment (SiggyF) [reply src/openclsim/model/move_activity.py]: Removed

File: src/openclsim/model/move_activity.py (hunk 2 above)
Comment (SiggyF) [reply src/openclsim/model/move_activity.py]: Removed

File: src/openclsim/core/movable.py [hunk 3]
```diff
@@ -20,60 +41,87 @@ class Movable(SimpyObject, Locatable):
...
-            )
+        if duration is None:
+            duration = self.compute_duration(self.geometry, destination.geometry)
 
         # Check out the time based on duration of sailing event
-        yield self.env.timeout(sailing_duration)
+        yield self.env.timeout(duration)
 
         # Set mover geometry to destination geometry
+        print("updating to destination geometry", destination.geometry)
```
[the remark this answers is not in the document]
Comment (SiggyF) [reply src/openclsim/core/movable.py]: Removed

File: src/openclsim/model/move_activity.py [hunk 4]
```diff
@@ -104,8 +106,9 @@ def main_process_function(self, activity_log, env):
             engine_order=self.engine_order,
             duration=self.duration,
         )
+        print("mover", self.mover.name, "is now at", self.mover.geometry.wkt)
```
[the remark this answers is not in the document]
Comment (SiggyF) [reply src/openclsim/model/move_activity.py]: removed

File: src/openclsim/core/movable.py [hunk 5]
```diff
@@ -126,11 +173,102 @@ def __init__(self, compute_v, *args, **kwargs):
...
             sum_level = self.container.get_level(id_)
             sum_capacity = self.container.get_capacity(id_)
         fill_degree = sum_level / sum_capacity
         return self.compute_v(fill_degree)
+
+
+class Navigator:
+    @staticmethod
+    def find_route(waypoints):
+        # TODO: implement this method
```
[the remark this answers is not in the document]
Comment (SiggyF) [reply src/openclsim/core/movable.py]: removed

File: src/openclsim/core/log.py [hunk 6]
```diff
@@ -53,11 +92,40 @@ def log_entry(
...
+            "ActivityID": activity_id,
+            "ActivityState": activity_state.name,
+            "ObjectState": object_state,
+            "ActivityLabel": activity_label,
+        }
+        self.logbook.append(entry)
+
+    def log_entry_v0(self, log, t: numbers.Number, value, geometry_log):
+        """Log an entry (opentnsim version)"""
+        assert isinstance(log, str), "expected log variable of type string"
```
[the remark this answers is not in the document]
Comment (SiggyF) [reply src/openclsim/core/log.py]: implemented typing consistently

File: notebooks/32_openclsim_opentnsim.ipynb [hunk 7]
```diff
@@ -0,0 +1,275 @@
+{
+ "cells": [
```
Comment (luke-moth) [inline notebooks/32_openclsim_opentnsim.ipynb:2]: we hebben nu 3 notebooks die met 32 beginnen...

File: .github/workflows/azure-static-web-apps-delightful-cliff-0e49c3503.yml [hunk 8]
```diff
@@ -32,7 +34,7 @@ jobs:
           # Install old version of pip /setuptools, issue with gdal. I still get this issue...
           python -m pip install --upgrade pip setuptools==57.5
           # install gdal separate
-          pip install wheel==0.38.4 gdal==3.0.4
+          pip install wheel==0.38.4 gdal==3.4.1
```
Comment (luke-moth) [inline .github/workflows/azure-static-web-apps-delightful-cliff-0e49c3503.yml:37]: fijn!

**Review by luke-moth — APPROVED** [review]
Ziet er goed uit, grotendeels vorige week al besproken natuurlijk.

**Pull request closed without merging**